src/main/scala/zio/s3/settings.scala

```diff
@@ -73,8 +73,13 @@ object S3Credentials {
       )
     )(p => effectBlocking(p.resolveCredentials()))
 
+  val fromWebIdentity: ZIO[Blocking, InvalidCredentials, S3Credentials] =
+    load(
+      ZManaged.effect(WebIdentityTokenFileCredentialsProvider.create())
+    )(p => effectBlocking(p.resolveCredentials()))
+
   val fromAll: ZIO[Blocking, InvalidCredentials, S3Credentials] =
-    fromSystem <> fromEnv <> fromProfile <> fromContainer <> fromInstanceProfile
+    fromSystem <> fromEnv <> fromProfile <> fromContainer <> fromInstanceProfile <> fromWebIdentity
```

Comment:
i would exclude this from the default resolver, since aws dont have this one by default, i try to follow the same behavior, also you are forcing by default to have st dependency.


https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html

```
  1.  In the Java system properties: aws.accessKeyId and aws.secretAccessKey.
  2.  In system environment variables: AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.
  3.  The default credential profiles file– The specific location of this file can vary per platform, but is typically located at ~/.aws/credentials
  4.  In the Amazon ECS environment variable: AWS_CONTAINER_CREDENTIALS_RELATIVE_URI.
  5.   In the instance profile credentials
```

Reply:
That is not properly documented because [they do support it](https://github.com/aws/aws-sdk-java-v2/pull/1501) At least in their interface they do. So what they are telling users is that they need to add the sts dependency on their project for it to work. what we can do here is the same. remove the sts dependency, keep it on the `fromAll` which mimics the `DefaultChainProvider`, and just tell users they need to add sts for it to work.

Reply:
@sken77 thanks for the feedback. SO in this case i would add by default the `sts` dependency in zio-s3 and you can remove the comment. 
I don't like the idea to have something failing if you are calling `fromAll` because of missing dependency.

Reply:
I see what you are saying. i read the code and it says that it would be fail and tell you that you need to add the STS dependency, which might be confusing for people not caring about that and failing due to other reasons.

Reply:
the nice thing about calling a combinator like `fromAll` is that you can test this in different instances (i.e your local, ec2, kube) without changing anything.

Reply:
What do you think of this proposals

1. remove `fromWebIdentity` from the alias `fromAll`
or
2. add `sts` dependency